src/azul/indexer/__init__.py

```diff
@@ -250,13 +253,10 @@ def parse(cls, spec: str) -> 'SimpleSourceSpec':
         ...
         azul.uuids.InvalidUUIDPrefixError: '//foo.edu' is not a valid UUID prefix.
         """
-
-        # FIXME: Move parsing of prefix to SourceSpec
-        #        https://github.com/DataBiosphere/azul/issues/3073
-        name, sep, prefix = spec.rpartition(':')
-        reject(sep == '',
-               'Source specifications must end in a colon followed by an optional UUID prefix')
-        return cls(prefix=Prefix.parse(prefix), name=name)
+        _spec, prefix = super().parse(spec)
+        self = cls(prefix=prefix, name=_spec)
+        assert spec == str(self), (spec, str(self), self)
```

Comment:
```suggestion
        assert spec == str(self), spec
```

Only include the minimum amount of info needed to reproduce.

Reply:
If this assertion fires I'm sure we would at least want to know what `self` was.